archivist_samples/door_entry/run.py

```diff
@@ -48,324 +47,246 @@
+def create_gdn_front(doors):
 
-    doors_create(
+    return doors_create(
         doors,
-        "front.gdn.paris.wavestonedas",
         "Gare du Nord apartments front door",
-        "das-x4-04",
-        (
-            "Electronic door entry system controlling the front "
-            "residential entrance to Apartements du Gare du Nord"
-        ),
-        location,
-        attachments,
+        {
+            "arc_firmware_version": "1.0",
+            "arc_serial_number": "das-x4-04",
+            "arc_description": (
+                "Electronic door entry system controlling the front "
+                "residential entrance to Apartements du Gare du Nord"
+            ),
+            "wavestone_asset_id": "front.gdn.paris.wavestonedas",
+        },
+        location={
+            "props": {
+                "display_name": "Apartements du Gare du Nord",
+                "description": (
+                    "Residential apartment building in new complex " "above GdN station"
+                ),
+                "latitude": 48.8809,
+                "longitude": 2.3553,
+            },
+            "attrs": {
+                "address": "18 Rue de Dunkerque, 75010 Paris, France",
+                "wavestone_ext": "managed",
+            },
+        },
+        attachments=[
+            "gdn_front.jpg",
+        ],
     )
 
 
-def create_gdn_side_image(doors):
-    with pkg_resources.open_binary(images_assets, "gdn_side.jpg") as fd:
-        return doors.attachments.upload(fd)
-
+def create_gdn_side(doors):
 
-def create_gdn_side(doors, location, attachments):
-
-    doors_create(
+    return doors_create(
         doors,
-        "side.gdn.paris.wavestonedas",
         "Gare du Nord apartments side door",
-        "das-x4-05",
-        (
-            "Electronic door entry system controlling the side "
-            "residential entrance to Apartements du Gare du Nord"
-        ),
-        location,
-        attachments,
+        {
+            "arc_firmware_version": "1.0",
+            "arc_serial_number": "das-x4-05",
+            "arc_description": (
+                "Electronic door entry system controlling the side "
+                "residential entrance to Apartements du Gare du Nord"
+            ),
+            "wavestone_asset_id": "side.gdn.paris.wavestonedas",
+        },
+        location={
+            "props": {
+                "display_name": "Apartements du Gare du Nord",
+                "description": (
+                    "Residential apartment building in new complex " "above GdN station"
+                ),
+                "latitude": 48.8809,
+                "longitude": 2.3553,
+            },
+            "attrs": {
+                "address": "18 Rue de Dunkerque, 75010 Paris, France",
+                "wavestone_ext": "managed",
+            },
+        },
+        attachments=[
+            "gdn_side.jpg",
+        ],
     )
 
 
 def create_doors(doors):
```

Comment:
Having a method called `doors_create` and a method called `create_doors` isn't super clear - perhaps one or both of them could be renamed to illustrate the different jobs they perform?

Reply:
renaming doors_create to 'doors_creator' would be acceptable?

Reply:
Yeah that sounds slightly better - maybe a comment / docstring above the function too?

Reply:
fixed here and else where